process/block/alteredaccounts/tokensProcessor.go

```diff
@@ -113,7 +129,8 @@ func (tp *tokensProcessor) extractEsdtData(
 		return err
 	}
 
-	if len(topics) > idxReceiverAddressInTopics {
+	eventShouldContainReceiverAddress := string(event.GetIdentifier()) != core.BuiltInFunctionESDTNFTCreate
```

Comment:
I think events with identifier ESDTNFTTransfer, ESDTTransfer, MultiTransfer And Wipe contains receiver address on the topics[3]

Reply:
true. refactored